### DISCUSSION TITLE
DeePMD-kit TensorBoard usage

### DISCUSSION THREAD

**Discussion by alib-l** [deepmodeling/deepmd-kit#420] [Q&A]
Hello, 
I set the Tensorboard tag and its log direction in my input file, however, it does not give the log folder in the output. How can I solve it?
Thanks,

**Comment by amcadmus** [comment]
@marian-code  could you please take a look? Thanks!

**Comment by marian-code** [comment]
The problem is a bit tricky since the tensorboard support has not been really put out in the wild. It is stuck on the `devel` branch so if you have any of the conda/docker/compiled from master version you obviously won't have tensorboard.

If you still want it:
* I would not suggest compiling from `devel` branch now since it is entangled with `api` branch that is in process of rather massive rewrites so there might be bugs.
* The best bet I think, if you really want to use it is to use the current `1.3.2` tag and use git cherry-pick to pull out the necessary commits from #322 . That should be rather straightforward, as there aren't that many changes.

Or maybe we should do this @amcadmus ? and put it out in `v1.3.3`

**Reply by amcadmus** [reply]
I would no suggest cherry-picking the tensorboard support to branch 1.3, which is maintained not for new features but for bug fixings.

**Reply by alib-l** [reply]
Thank you,
I just want to visualize the trained model, and also plot the pairwise potential. How can I do that instead of using Tensorboard?

**Reply by marian-code** [reply]
Well, there is nothing to visualize. There is no pairwise potential in sense of say, lennard-jones. This is really a multi body potential and there is no way to 'visualize' it. I think unfortunately even tensorboard would not give you the insight you are hoping for. The main things you would see there are the computational graph schematic and model weights but that provides little insight in a physical sense that as I understand you are hoping to gain. This is really one of the drawbacks of NN potentials that you loose this physical insight and have to take it only as black-box that when trained properly does what it is supposed to.

Hope this helps to clear things up.

**Reply by alib-l** [reply]
Thank you,

**Reply by ASaydam** [reply]
I believe tensorboard support did not reach to master branch yet. Is it coming soon?
It took me a while to find this Q&A and I tried to run it using the https://deepmd.readthedocs.io/ instructions without success.
